model-optimizer/extensions/analysis/tf_od_api.py

```diff
@@ -16,62 +16,83 @@ class TensorFlowObjectDetectionAPIAnalysis(AnalyzeAction):
                               'SecondStageBoxPredictor',
                               'SecondStageBoxPredictor/map',
                               'SecondStageBoxPredictor/map_1',
                               'SecondStagePostprocessor',
-                              ]),
-                    ('FasterRCNN', ['Preprocessor',
-                                    'FirstStageFeatureExtractor',
+                              ],),
+                    'FasterRCNN': (['FirstStageFeatureExtractor',
                                     'SecondStageFeatureExtractor',
                                     'SecondStageBoxPredictor',
                                     'SecondStagePostprocessor',
-                                    ]),
-                    ('SSD', ['Preprocessor',
-                             'FeatureExtractor',
+                                    ],
+                                   ['FirstStageRPNFeatures',
+                                    'FirstStageBoxPredictor',
+                                    'SecondStagePostprocessor',
+                                    'mask_rcnn_keras_box_predictor',
+                                    ],),
+                    'SSD': (['FeatureExtractor',
                              'Postprocessor',
-                             ]),
-                    ]
-
-    file_patterns = {'MaskRCNN': 'mask_rcnn_support.*\\.json',
-                     'RFCN': 'rfcn_support.*\\.json',
-                     'FasterRCNN': 'faster_rcnn_support.*\\.json',
-                     'SSD': 'ssd.*_support.*\\.json',
-                     }
+                             ],
+                            ['ssd_mobile_net_v2keras_feature_extractor',
+                             'Postprocessor'],
+                            ['ssd_mobile_net_v1fpn_keras_feature_extractor',
+                             'Postprocessor'],
+                            ['ssd_mobile_net_v2fpn_keras_feature_extractor',
+                             'Postprocessor'],
+                            ['ResNet50V1_FPN',
+                             'Postprocessor'],
+                            ['ResNet101V1_FPN',
+                             'Postprocessor'],
+                            ['ResNet152V1_FPN',
+                             'Postprocessor']
+                            ),
+                    }
 
     def analyze(self, graph: Graph):
-        if any([name not in graph.nodes() for name in ['image_tensor', 'detection_classes', 'detection_boxes',
-                                                       'detection_scores']]):
+        tf_1_names = ['image_tensor', 'detection_classes', 'detection_boxes', 'detection_scores', 'Preprocessor']
+        tf_1_cond = all([any([node.soft_get('name').find(scope) != -1 for node in graph.get_op_nodes()])
+                         for scope in tf_1_names])
+
+        tf_2_names = ['input_tensor', 'output_control_node', 'Identity', 'Preprocessor']
```

Comment:
I am not sure that this list of names are really representative for all TF2 OD models because `Identity` presents in majority TF2 models not only OD.
There can be either `Preprocessor` or `map` names as well